### PR TITLE
Add userAssignedIdentities

### PR DESCRIPTION
**Pull request by chidmdxx**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by annatisch** [comment]
Thanks @chidmdxx! The changes in the PR look good.

However this spec has a large number of ARM violations in the linter CI. They are all unrelated to the change in this PR however they should be addressed. A large number of them could easily be resolved with some improved descriptions or adding examples. Others could be resolved by applying for suppression until a longer term fix can be implemented.
https://travis-ci.org/Azure/azure-rest-api-specs/jobs/394813179

Would it be possible to address any of these?

**Comment by annatisch** [comment]
ping @chidmdxx

**Comment by annatisch** [comment]
Thanks @chidmdxx - please contact @ravbhatnagar to apply for suppressions. :)

File: specification/resources/resource-manager/Microsoft.Resources/stable/2018-05-01/resources.json [hunk 1]
```diff
@@ -2952,6 +2952,11 @@
             "name": "ResourceIdentityType",
             "modelAsString": false
           }
+        },
+        "userAssignedIdentities":{
```
Comment (ravbhatnagar) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2018-05-01/resources.json]: This object has a defined shape correct? Can we add the complete schema for this?

Comment (chidmdxx) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2018-05-01/resources.json]: there is no defined shape
it's a dictionary where keys are the resource ids and the value is an empty jobject

**Comment by ravbhatnagar** [comment]
Discussed offline. Added an example for the shape of the object.

**Comment by ravbhatnagar** [comment]
@annatisch @chidmdxx - If the violations are something we will fix, there is no suppression process. We are tracking towards fixing these. This should not block the PR.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2018-05-01/resources.json (hunk 1 above)
Comment (chidmdxx) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2018-05-01/resources.json]: Added the example file as discussed

Comment (anuchandy) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2018-05-01/resources.json]: Can we have something similar to what compute did https://github.com/Azure/azure-rest-api-specs/blob/6bc56bbc8f9a137a57692bec82cce62ed8579868/specification/compute/resource-manager/Microsoft.Compute/stable/2018-06-01/compute.json#L5642

Comment (anuchandy) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2018-05-01/resources.json]: Customer s will have hard time using the SDK without having a shape defined for such fields

Comment (chidmdxx) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2018-05-01/resources.json]: thanks @anuchandy was having a hard time with how to shape the swagger property

**Comment by annatisch** [comment]
@chidmdxx - could you please let me know you intentions regarding the remaining violations? If you intend to fix them in subsequent PRs I will go ahead and merge this one.

**Comment by chidmdxx** [comment]
@annatisch as @ravbhatnagar mentioned we are tracking in ARM to fix the violation is subsequent PRs

**Review by annatisch — APPROVED** [review]

**Pull request merged**